websocketOutportDriver/sender/webSocketSender.go

```diff
@@ -214,61 +196,40 @@ func (w *webSocketSender) waitForAck(remoteAddr string, counter uint64) {
 			return
 		}
 
-		_, ok = acksForAddress[counter]
+		ok = acksForAddress.ProcessAcknowledged(counter)
 		if ok {
-			w.removeAcknowledge(remoteAddr, counter)
 			return
 		}
 
 		time.Sleep(time.Millisecond)
 	}
 }
 
-func (w *webSocketSender) getAcknowledges() map[string]map[uint64]struct{} {
+func (w *webSocketSender) getAcknowledges() map[string]*websocketClientAcknowledgesHolder {
 	w.mutAcknowledges.RLock()
 	defer w.mutAcknowledges.RUnlock()
 
 	return w.acknowledges
 }
 
-func (w *webSocketSender) addCounterToAcknowledges(remoteAddr string, counter uint64) {
-	w.mutAcknowledges.Lock()
-	_, exists := w.acknowledges[remoteAddr]
-	if exists {
-		w.acknowledges[remoteAddr][counter] = struct{}{}
-		w.mutAcknowledges.Unlock()
-
-		return
-	}
-
-	w.log.Warn("adding counter to non-existing remote addr", "remote addr", remoteAddr, "counter", counter)
-	w.acknowledges[remoteAddr] = make(map[uint64]struct{}) // should never reach here
-}
-
-func (w *webSocketSender) removeAcknowledge(remoteAddr string, counter uint64) {
-	// for a better performance (avoid existing checks), this function relies on the fact that the value exists in the map
-	w.mutAcknowledges.Lock()
-	acks := w.acknowledges[remoteAddr]
-	delete(acks, counter)
-	w.mutAcknowledges.Unlock()
-}
-
-// SendOnRoute will make the request accordingly to the received arguments
-func (w *webSocketSender) SendOnRoute(args data.WsSendArgs) error {
+// Send will make the request accordingly to the received arguments
+func (w *webSocketSender) Send(args data.WsSendArgs) error {
+	w.mutCounterOperations.Lock()
```

Comment:
not a great solution. Instead of mutCounterOperations and counter we could have used an uint64 and `atomic` package
Please try this code:
```GO
    var counter uint64

    fmt.Println(atomic.AddUint64(&counter, 1)) // safely add 1 and return the new value -> returns 0+1 
    fmt.Println(atomic.LoadUint64(&counter))   // safely returns stored value -> returns 1
    fmt.Println(atomic.AddUint64(&counter, 1)) // safely add 1 and return the new value -> return 1+1
```

Reply:
done